lib/identity_cache/should_use_cache.rb

```diff
@@ -0,0 +1,13 @@
+module IdentityCache
+  module ShouldUseCache
+    extend ActiveSupport::Concern
+
+    included do
+      unless self.respond_to?(:should_use_cache?)
+        def self.should_use_cache?
```

Comment:
Including a module is like multiple inheritance, so it looks for the method in the class itself before looking for it in the included modules.  E.g.

```ruby
module Mixin
  def print
    puts 'mixin'
  end
end

class Thing
  def print
    puts 'thing'
  end
  include Mixin
end

Thing.new.print
```

prints `thing` because it found Thing#print before looking for Mixin#print.  However, what you were doing was basically just replacing Thing#print when `include Mixin` is called.